protostar/formatter/formatting_result.py

```diff
@@ -1,82 +1,96 @@
-from typing import List
+from abc import ABC
 from dataclasses import dataclass
 from pathlib import Path
 
-from protostar.io.log_color_provider import log_color_provider
+from protostar.io import Message, LogColorProvider
 
 
 @dataclass
-class FormattingResult:
+class FormattingResult(Message, ABC):
     filepath: Path
+    check: bool
```

Comment:
Why is `check` needed here?

Reply:
1. I need it to determine what header to use for correct & incorrect results (mind that these data structures are now self-containing, they have all information they need for presentation and they do not have to reference "parents")
2. I would love to put `check` only in correct & incorrect and skip it for `broken`, but padding of headers essentially forces me to make `check` more general because it is needed in each variant.

Reply:
`Check` is an input parameter. The name `check` doesn't make sense within result's context.

Reply:
It actually does! It's good to know from the result whether the operation modified stuff or not IMHO

Reply:
done